### DISCUSSION TITLE
How to customize recharts-active-dot?

### DISCUSSION THREAD

**Discussion by summerisgood1** [recharts/recharts#4358] [Q&A] [answered]
![Screenshot 2024-03-28 at 15 12 37](https://github.com/recharts/recharts/assets/51513183/db866a89-2e73-4753-9068-c750273da4d0)

I can set display none on .recharts-active-dot nut i cannot set the color or background color.

**Comment by ckifer** [comment, marked as answer]
Hi - recharts uses SVGs meaning styling is done via svg properties rather than CSS properties - use `stroke` and `fill` rather than `color` and `backgroundColor`

```tsx
      <Line
        type="monotone"
        dataKey="pv"
        stroke="#8884d8"
        activeDot={{ r: 8, fill: "#111", stroke: "blue" }}
      />
```

**Reply by ckifer** [reply]
If you want to use css classes then

`activeDot={{ r: 8, className: "active-dot" }}`

```css
.active-dot {
  stroke: red;
  fill: blue;
}
```
![image](https://github.com/recharts/recharts/assets/25180830/37279621-8ef8-4524-9ed9-a2c0e39c5b53)

**Comment by Sourabh-jana** [comment]
how to have diffrent shapes in dot

**Reply by ckifer** [reply]
You can do something like this 

```
      <Line
        type="monotone"
        dataKey="uv"
        stroke="#82ca9d"
        activeDot={(props: DotProps) => <CustomizedDot {...props} />}
      />
```

https://codesandbox.io/p/sandbox/line-chart-with-customized-dot-forked-s96f9y
